README.md

```diff
@@ -4,252 +4,85 @@
   <a href="https://github.com/actions/setup-java"><img alt="GitHub Actions status" src="https://github.com/actions/setup-java/workflows/Main%20workflow/badge.svg"></a>
 </p>
 
-This action sets up a java environment for use in actions by:
-
-- optionally downloading and caching a requested version of java by version and adding to PATH. Default downloads are populated from the [Zulu Community distribution of OpenJDK](http://static.azul.com/zulu/bin/)
-- registering problem matchers for error output
-
-# Usage
-
-See [action.yml](action.yml)
-
-## Basic
+This action provides the following functionality for GitHub Actions runners:
+- Downloading and setting up a requested version of Java. See [Usage](#Usage) for a list of supported distributions
+- Extracting and caching custom version of Java from a local file
+- Configuring runner for publishing using Apache Maven
+- Configuring runner for publishing using Gradle
+- Configuring runner for using GPG private key
```

Comment:
This action should be reconfigured so other movements within the program can run synbchronously